runtime/compiler/control/CompilationThread.cpp

```diff
@@ -247,7 +264,125 @@ jitSignalHandler(struct J9PortLibrary *portLibrary, U_32 gpType, void *gpInfo, v
 
    return J9PORT_SIG_EXCEPTION_CONTINUE_SEARCH;
    }
-
+#ifdef COMPRESS_AOT_DATA
+#ifdef J9ZOS390
+#pragma convlit(suspend)
+#endif
+/*
+ * \brief
+ *    Inflates the data buffer using zlib into output buffer
+ *
+ * \param 
+ *    buffer Input buffer that is going to be inflated 
+ * 
+ * \param 
+ *    numberOfBytes Size of the data in the input buffer to inflate
+ * 
+ * \param
+ *    outBuffer Output buffer to hold the deflated data
+ * 
+ * \param
+ *    unCompressedSize Size of the original data
+ * 
+ * \return
+ *    Returns DECOMPRESSION_FAILED if it can not inflate the buffer
+ * 
+ */
+static
+int inflateBuffer(U_8 *buffer, int numberOfBytes, U_8 *outBuffer, int unCompressedSize)
+   {
+   Bytef *in = (Bytef*)buffer;
+   Bytef *out = (Bytef*)outBuffer;
+   z_stream _stream;
+   _stream.zalloc = Z_NULL;
+   _stream.zfree = Z_NULL;
+   _stream.opaque = Z_NULL;
+   _stream.avail_in = 0;
+   _stream.next_in = Z_NULL;
+   int ret = inflateInit(&_stream);
+   if (ret != Z_OK)
+      return DECOMPRESSION_FAILED;
+   _stream.avail_out = unCompressedSize;
+   _stream.next_out = out;
+   _stream.next_in = in;
+   _stream.avail_in = numberOfBytes;
+   ret = inflate(&_stream, Z_NO_FLUSH);
+   switch (ret)
+      {
+      case Z_NEED_DICT:
+      case Z_DATA_ERROR:
+      case Z_MEM_ERROR:
+         inflateEnd(&_stream);
+         return DECOMPRESSION_FAILED;
+      }
+   ret = DECOMPRESSION_SUCCEEDED;
+   if (_stream.avail_out != 0)
+      {
+      TR_ASSERT(0, "Fails while inflating buffer");
+      ret = DECOMPRESSION_FAILED;
+      }
+   inflateEnd(&_stream);
+   return ret;
+   }
+
+/*
+ * \brief
+ *    Deflates the data buffer using zlib into output buffer
+ *
+ * \param 
+ *    buffer Input buffer that is going to be deflated 
+ * 
+ * \param 
+ *    numberOfBytes Size of the data in the input buffer to deflate
+ * 
+ * \param
+ *    outBuffer Output buffer to hold the deflated data
+ * 
+ * \param
+ *    level Level of compression
+ * 
+ * \return
+ *    If successful, returns Size of data in the deflated bufer
+ *    else returns COMPRESSION_FAILED
+ * 
+ */
+static
+int deflateBuffer(const U_8 *buffer, int numberOfBytes, U_8 *outBuffer, int level)
+   {
+   z_stream _stream;
+   _stream.zalloc = Z_NULL;
+   _stream.zfree = Z_NULL;
+   _stream.opaque = Z_NULL;
+   int ret=deflateInit(&_stream, level);
+   if (ret != Z_OK)
+      return COMPRESSION_FAILED;
+   // Start deflating
+   _stream.avail_in = numberOfBytes;
+   _stream.next_in = (Bytef*) buffer;
+   _stream.avail_out = numberOfBytes;
+   _stream.next_out = (Bytef*) outBuffer;
+   ret = deflate(&_stream, Z_FINISH);
+   /**
+    * ZLIB return Z_STREAM_END if the buffer streamed to be deflated is finished.
```

Comment:
There are some typos here:
return --> returns
streamed --> stream

Reply:
Fixed in https://github.com/eclipse/openj9/pull/5926/commits/66c4368333f9f5ba388b64ec16893d59d9c7eaa1
And 
https://github.com/eclipse/openj9/pull/5926/commits/db11cf72649c89047724255e7f0c1791efe77b3a